feat(useful-packages): Replace `moment.js` with `Day.js`. (#685)
discordjs/guide: guide/miscellaneous/useful-packages.md

# Useful packages

## Day.js

::: tip
Official documentation: [https://day.js.org/](https://day.js.org/en)
:::

Day.js is a powerful package that parses, validates, manipulates, and displays dates and times in JavaScript.
It allows you to quickly and easily format dates in any way you want or parse strings back into JavaScript Date objects.  

> There are even some plugins for it to allow you to work with durations and more.

Reviewer: Day.js without any plugin is poor on features, so you should say that it often requires plugins.

Reviewer 2: Surely a better option would be to use a lib such as date-fns? Obviously, there are many different date-time libraries, but I've personally found date-fns to be one of the better ones.

Author: Had talked with @Danktuary regarding this earlier, and we decided to use `Day.js` in the guide.

Reviewer 3: For the purposes and examples listed in this page, I think day.js is fine. It has a simple API, is small in size, and has plugins for common use cases.

For example if you wanted to ask your users to give you a date,  
you can use Day.js to turn it into a Date object you can use in your code:

<!-- eslint-skip -->
```js
const input = await message.channel.awaitMessages(m => m.author.id === message.author.id, {
	max: 1,
	time: 10e3,
	errors: ['time'],
});
const date = dayjs(input.first().content).toDate();
```

> Using the "Duration" plugin, you could tell the user how many days in the future or past the date is:

Reviewer 3: ```suggestion
Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user how many days in the future or past the date is:
```

> Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user how many days in the future or past the date is:

Reviewer 3: ```suggestion
Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user if the date is in the future or the past:
```
Didn't notice this at first but this sentence doesn't exactly line up with the codeblock below.
